parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1171,6 +1188,16 @@ public ReadBuilder withNameMapping(NameMapping newNameMapping) {
       return this;
     }
 
+    @Override
+    public ReadBuilder setRootType(Class<? extends StructLike> rootClass) {
+      throw new UnsupportedOperationException("Custom types are not yet supported");
```

Comment:
When the internal object model is complete, this should be implemented to instantiate the expected metadata classes while reading.